microsoft-edge/devtools-guide-chromium/memory-problems/index.md

```diff
@@ -281,18 +276,197 @@ DevTools now tracks all of the objects that were GC'd during the recording.  Use
+
+   The demo webpage begins generating messages and displaying them in the webpage:
+
+   ![Generating some messages in the demo webpage](./index-images/generate-messages.png)
+
+1. After some messages are displayed, click the **Stop** button in the demo webpage.
+
+   Each message is a `<div class="message">` element that's referenced by the Room 1 instance of the `Room` class.  There are no detached elements in the webpage DOM tree, because all of the message elements are attached to the present, Room 1 instance of the **Room** class.
+
+
+   <!-- ------------------------------ -->
+   **Change to a different instance of the Room class, so elements become detached:**
+
+1. In the demo webpage, click the **Room 2** button, which corresponds to the Room 2 instance of the `Room` class.
+
+   In the webpage, the messages disappear:
+
+   ![Room 2 initial view](./index-images/room-2-initial-view.png)
+
+   The messages that were generated for the Room 1 instance of the **Room** class (`<div class="message">` elements) are no longer attached to the DOM, but they're still referenced by the Room 1 instance of the **Room** class.  They are detached elements, which are a memory leak, unless they are going to be used again by the webpage.
+
+
+   <!-- ------------------------------ -->
+   **Get the list of detached elements:**
+
+1. In DevTools, in the **Memory** tool, select the **Detached elements** option button (profiling type), and then click the **Start** button.
+
+   The list of detached nodes is displayed in the **Detached nodes** column of the generated **Detached elements** profile:
+
+   ![List of detached nodes in the Detached nodes column of the generated Detached elements profile](./index-images/detached-dom-nodes.png)
+
+   The detached elements are displayed, as DOM nodes like in the DOM tree in the **Elements** tool.  In the **Profiles** list, in the **Detached elements** section, the profile is listed.
+
+1. Expand a DOM node to display its child elements (DOM nodes):
+
+   ![An expanded detached DOM node](./index-images/detached-dom-node-expanded.png)
+
+   These detached elements (DOM nodes) are memory leaks, if they aren't going to be reused by the application.
+
+
+If you want to return to the list of option buttons for profiling types, in the upper left of the **Memory** tool, click **Profiles** (![the Profiles icon](./index-images/profiles-icon.png)).
+
+For additional ways to assess memory leaks, see [Tools for investigating detached elements](#tools-for-investigating-detached-elements), above.
+<!-- 
+See also:
+* [Detached Elements profiling type in Memory tool](../whats-new/2024/10/devtools-130.md#detached-elements-profiling-type-in-memory-tool) in _What's New in DevTools (Microsoft Edge 130)_.
+-->
+
+
+<!--[1]-->
 <!-- ====================================================================== -->
-## Spot frequent garbage collections
+## Find DOM tree memory leaks ("Heap snapshot" profiling type > Detached)
```

Comment:
how to you get to the JavaScript source code from a detached node at the end of this section?

rendered article section:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/?branch=pr-en-us-3282#find-dom-tree-memory-leaks-heap-snapshot-profiling-type--detached

@captainbrosset

Reply:
Maybe replace these instructions below with instructions that use the Detached Elements demo instead. Using that demo, the steps are roughly:
1. Room 1, Fast traffic button, Stop button. Then Room 2.
2. GC (trashcan icon).
3. Record heap snapshot.
4. Click **Filter by class** and type **Detached**.
5. Expand **Detached <div>** object in the heap snapshot.
6. Select one of the **<div class="message">** in the list.
    Link to soure code is displayed in the **Retainers** panel, at the bottom.

Reply:
docs design: at this point, it's unclear whether this h2 section (Heap snap > Detach) will remain in Mem tool article; almost certainly not.  Predict this section's content will be moved either to:
* Heap shap prof type article <-- probably here, b/c articles are per prof type - despite section being a crossover of heap snap & detached, it's literally using the HS prof type therefore section should move to there, not to DE prof type article.
* Detached elements prof type article

and will eventually remove this h2 heading from Mem tool article, and will link from Mem tool tools-overview table to h2 in one of the above dedicated articles, probably HS prof type article.
@captainbrosset